File: plugin/testDataOutput2/shared/extensions.swift
//Package: com.test
//Converted using Kwift2

import Foundation



public protocol SomeInterface {
}
 
 

extension Int32 {
    public func extensionPlusOne() -> Int32 {
        return self + 1
    }
}
 
 

extension Array {
    public func estimateWork() -> Int32 {
        return self.size
    }
}
 
 

extension Array where OVERRIDE == SomeInterface {
    public func estimateWork2() -> Int32 {
        return self.size
    }
}
 
 

extension Array where OVERRIDE: SomeInterface {
    public func estimateWork3() -> Int32 {
        return self.size
    }
}
 
 

extension Array where S ==  SomeInterface {
    public func estimateWork4() -> Int32 {
        return self.size
    }
}
 
 

extension Array where S:  SomeInterface {
    public func estimateWork5() -> Int32 {
        return self.size
    }
}
 
 

extension Array {
    public func estimateWork<A>(method: A, how: (Element, A) -> Int32) -> Int32 {
        var amount = 0
        
        for item in self {
            amount += how(item, method)
        }
        return amount
    }
    public func estimateWork<A>(_ method: A, _ how: (Element, A) -> Int32) -> Int32 {
        return estimateWork(method: method, how: how)
    }
}
 

File: plugin/testData/shared/extensions.kt
package com.test

interface SomeInterface {}

fun Int.extensionPlusOne(): Int {
    return this + 1
}

fun <Element> List<Element>.estimateWork(): Int {
    return this.size
}

fun List<@swiftExactly("OVERRIDE") SomeInterface>.estimateWork2(): Int {
    return this.size
}

fun List<@swiftDescendsFrom("OVERRIDE") SomeInterface>.estimateWork3(): Int {
    return this.size
}

fun <S: @swiftExactly("OVERRIDE") SomeInterface> List<S>.estimateWork4(): Int {
    return this.size
}

fun <S: @swiftDescendsFrom("OVERRIDE") SomeInterface> List<S>.estimateWork5(): Int {
    return this.size
}

fun <Element, A> List<Element>.estimateWork(method: A, how: (Element, A)->Int): Int {
    var amount = 0
    for(item in this){
        amount += how(item, method)
    }
    return amount
}
